frontend/src/components/ActivityList.vue

```diff
@@ -201,14 +204,14 @@ export default {
           }
         });
     },
-    openItem(item) {
+    openItem(item: ActivityItem) {
       // if we click on it, mark it seen...
       this.$store.commit("activityNotificationSeen", { id: item.id });
       this.$store.commit("taskNotificationSeen", { id: item.id });
 
       console.log(item);
 
-      switch (item.type) {
+      switch (item.type.toUpperCase()) {
```

Comment:
ActivityItem and ActivityType are types generated from the backend. ActivityTypes (note the 's') is a frozen object from constants.ts which contains value and label keys (including i18n texts). In the switch-case statement the backend types sadly cannot be used like enums - therefore the ActivityTypes constant object is used here for convenience. I will add types to the ActivityTypes object. The constants might need some refactoring too since we now have TypeScript language features such as proper enums.
Reason for the toUpperCase() is that the backend returns the strings as lowercase, although the backend type defines it as uppercase.

Reply:
Someone started using lower case enums in the frontend, so to keep things consistent I also changed the backend so that everything is serialised lower case. I guess this can be changed at some point, but it will probably be some hassle to revert this.